src/main/java/uk/gov/di/ipv/cri/common/library/service/ConfigurationService.java

```diff
@@ -37,33 +41,52 @@ public enum SSMParameterName {
 
     @ExcludeFromGeneratedCoverageReport
     public ConfigurationService() {
-        this.clock = Clock.systemUTC();
-        this.ssmProvider = ParamManager.getSsmProvider();
-        this.secretsProvider = ParamManager.getSecretsProvider();
-        this.parameterPrefix =
-                Objects.requireNonNull(
-                        System.getenv("AWS_STACK_NAME"), "env var AWS_STACK_NAME required");
+        this(
+                ParamManager.getSsmProvider(
+                        SsmClient.builder().httpClient(UrlConnectionHttpClient.create()).build()),
+                ParamManager.getSecretsProvider(
+                        SecretsManagerClient.builder()
+                                .httpClient(UrlConnectionHttpClient.create())
+                                .build()),
+                getEnvironmentVariableValue("AWS_STACK_NAME"),
+                Optional.ofNullable(System.getenv("SECRET_PREFIX"))
+                        .orElse(getEnvironmentVariableValue("AWS_STACK_NAME")),
+                Clock.systemUTC());
     }
 
     public ConfigurationService(
```

Comment:
If this constructor isn't used anywhere other than unit tests then remove the `public` access modifier can be removed